Include archiveless posts in a post_status any query

srtfisher opened the pull request:
- Ensure that when `any` is passed as the `post_status` that archiveless does not modify the query. The query should return archiveless posts normally then with an unmodified query.
- Call out that `get_posts()` calls will not have archiveless posts returned by default. This logic in the plugin broke backward compatibility with some existing plugin use.

mboynes commented on inc/class-archiveless.php:
```diff
@@ -330,14 +330,15 @@ public function on_pre_get_posts( $query ) {
+		}
+
+		// Ignore any query for archiveless post specifically.
+		if ( ! empty( $query->get( 'post_status' ) ) && self::$status === $query->get( 'post_status' ) ) {
```
🤔 this seems redundant... right?

srtfisher replied:
True, removed!

mboynes commented:
why don't we want to modify the query if the post_status is set to `any`? and more importantly, will `any` include the archiveless status? reading my own code and not remembering it in-depth (😅), it seems like `any` might not include it, so I was explicitly adding it 🤷

srtfisher commented:
@mboynes this is more so for non-main queries. Before #39 this was using post_where and wasn't modifying non-main queries at all (see the old code here https://github.com/alleyinteractive/archiveless/blob/before-refactor/class-archiveless.php#L307-L317). Now we do need to modify almost every query unless they're for a specific post_status.

We totally don't need the secondary check for 'post status != archiveless', I removed that.

The root of my problem was trying to query for all posts (including archiveless posts) through a non-main query. I set the `post_status = any` and I still didn't receive the expected posts. My mistake was assuming that excluding `any` would fix it but it was just a red herring. The actual fix I needed to make was in `get_default_post_statuses()`. 

I was setting `publicly_queryable` to true which core doesn't do. Once that was removed, my custom query included archiveless posts again.

I've added a few tests to help test all the various edge cases, too.

srtfisher commented:
Coming back to this, I did walk back some of the previous changes made to this plugin. With #39 it would start including archiveless posts with any `get_posts()` call. This breaking change presented some problems with existing usage of the archiveless plugin. 

I updated this plugin to only ignore `post_status` set to `any` and not `publish`. I also clarified the documentation https://github.com/alleyinteractive/archiveless/tree/post-status-any#handling-archiveless-posts-with-get_posts-calls to also call out that `get_posts()` calls will not have archiveless posts in them by default.

Added some more tests as well.

mogmarsh commented on tests/test-general.php:
```diff
@@ -91,12 +110,85 @@ public function test_archiveless_is_method() {
+	}
+
+	/**
+	 * Test that archiveless posts are included in get_posts() calls by default.
```
```suggestion
	 * Test that archiveless posts are not included in get_posts() calls by default.
```
I think I'm reading everything right...

mogmarsh reviewed (approved):
one typo maybe? otherwise, 👍
